skymap_scanner/recos/millipede_original.py

```diff
@@ -159,9 +163,20 @@ def LatePulseCleaning(frame, Pulses, Residual=3e3*I3Units.ns):
         return ExcludedDOMs + [MillipedeOriginal.pulsesName_cleaned+'TimeWindows']
 
 
+    @staticmethod
     @icetray.traysegment
     def traysegment(tray, name, logger, seed=None):
         """Perform MillipedeOriginal reco."""
+        datastager = MillipedeOriginal.init_datastager()
+
+        abs_spline = datastager.get_filepath(MillipedeOriginal.MIE_ABS_SPLINE)
+        prob_spline = datastager.get_filepath(MillipedeOriginal.MIE_PROB_SPLINE)   
+
+        cascade_service = photonics_service.I3PhotoSplineService(abs_spline, prob_spline, timingSigma=0.0)
```

Comment:
Maybe to clarify my previous comment, I'm reading this as creating a new `I3PhotoSplineService` each time `traysegment` is run, while #140 took advantage of the fact that the `cascade_service` was created once at instantiation and then shared across processes. Isn't this going to duplicate the memory under parallelization?

Reply:
I believe our conclusion is that we're going to try to leverage `/dev/shm`. Mounting is platform dependent, but that's not the responsibility of the client code. Perhaps @dsschult can elaborate (https://icecube-spno.slack.com/archives/GU72UBBC3/p1681835217915089)?

Reply:
It should be no problem to support such a scenario if instantiate the reco class and we set up the photospline services as attributes "on demand". I would like to move away from the current "all static" paradigm anyway, so I will update this PR.

It is not clear to me whether `ewms-pilot` multitasking will allow to use shared memory somehow. My understanding is that now a subprocess(?) is spawned for every pixel to be reconstructed. But this is a different topic altogether.

Reply:
Yes, the idea was to use `/dev/shm` shared memory, since it's allocating it anyway.  The key will be to correctly handle the loading/cleanup so we don't get a deadlock situation.  Not sure if that needs to be included here, or wait until #194?

Reply:
With the last changes the reconstruction class is instantiated in `reco_icetray.py`. It provides a `setup_reco()` method that stages the spline data and initializes the photospline services within the object. The reco traysegment is now a bound method.

This should allow more flexibility and better control re to resource allocation. Cannot really foresee whether the design is compatible with `ewms-pilot` multiprocessing but we can re-evaluate if necessary.

Reply:
Right, for #148, I'll test this in different environments (container, non-container, condor) and measure memory usage.